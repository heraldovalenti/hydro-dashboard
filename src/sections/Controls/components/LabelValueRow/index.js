import React from 'react';
import { Input, InputAdornment } from '@material-ui/core';

export default function LabelValueRow({
  label,
  type,
  value,
  setValue,
  adornmentType,
}) {
  function handleValueChange(event) {
    const value = type === 'number' ? +event.target.value : event.target.value;
    setValue(value);
  }

  return (
    <div className="row row--label-value">
      <div className="label">{label}</div>
      <Input
        className="input-field"
        onChange={handleValueChange}
        InputLabelProps={{
          shrink: true,
        }}
        inputProps={{
          min: 0,
          type,
        }}
        value={value}
        startAdornment={
          adornmentType === 'money' && (
            <InputAdornment position="start">$</InputAdornment>
          )
        }
        endAdornment={
          adornmentType === 'percent' && (
            <InputAdornment position="end">%</InputAdornment>
          )
        }
      />
    </div>
  );
}
