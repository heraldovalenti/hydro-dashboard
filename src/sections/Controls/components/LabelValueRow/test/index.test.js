import React from 'react';
import LabelValueRow from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('<LabelValueRow/>', () => {
  it('renders default', () => {
    const testProps = {
      label: '',
      value: '',
      setValue: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<LabelValueRow {...testProps} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
