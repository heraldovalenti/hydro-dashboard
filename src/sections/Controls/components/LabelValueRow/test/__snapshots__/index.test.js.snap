// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelValueRow/> renders default 1`] = `
<div
  className="row row--label-value"
>
  <div
    className="label"
  >
    
  </div>
  <ForwardRef(WithStyles)
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    className="input-field"
    endAdornment={false}
    inputProps={
      Object {
        "min": 0,
        "type": undefined,
      }
    }
    onChange={[Function]}
    startAdornment={false}
    value=""
  />
</div>
`;
